diff -Nur asciidoc-8.6.6/asciidoc.py asciidoc-8.6.6-new/asciidoc.py
--- asciidoc-8.6.6/asciidoc.py	2011-09-03 01:04:19.000000000 +0400
+++ asciidoc-8.6.6-new/asciidoc.py	2011-11-02 22:02:37.359198461 +0400
@@ -785,11 +785,14 @@
         if not found:
             if USER_DIR:
                 found = findfilter(filtername, USER_DIR, cmd)
-            if not found:
+        if not found:
+	    found = findfilter(filtername, DATA_DIR, cmd)
+	    if not found:
                 if localapp():
                     found = findfilter(filtername, APP_DIR, cmd)
                 else:
                     found = findfilter(filtername, CONF_DIR, cmd)
+                    
     else:
         if os.path.isfile(cmd):
             found = cmd
@@ -5800,6 +5803,7 @@
 USER_DIR = None             # ~/.asciidoc
 # Global configuration files directory (set by Makefile build target).
 CONF_DIR = '/etc/asciidoc'
+DATA_DIR = '/usr/share/asciidoc'
 HELP_FILE = 'help.conf'     # Default (English) help file.
 
 # Globals
diff -Nur asciidoc-8.6.6/Makefile.in asciidoc-8.6.6-new/Makefile.in
--- asciidoc-8.6.6/Makefile.in	2011-11-02 22:04:15.000000000 +0400
+++ asciidoc-8.6.6-new/Makefile.in	2011-11-02 21:45:39.703402427 +0400
@@ -23,6 +23,7 @@
 VPATH = @srcdir@
 
 ASCIIDOCCONF = $(sysconfdir)/asciidoc
+ASCIIDOCDATA = $(datadir)/asciidoc
 
 prog = asciidoc.py a2x.py
 progdir = $(bindir)
@@ -35,25 +36,26 @@
 conf = $(wildcard *.conf)
 confdir = $(ASCIIDOCCONF)
 
-filtersdir = $(ASCIIDOCCONF)/filters
+filtersdir = $(ASCIIDOCDATA)/filters
+filtersconfdir = $(ASCIIDOCCONF)/filters
 
 codefilter = filters/code/code-filter.py
 codefilterdir = $(filtersdir)/code
 codefilterconf = filters/code/code-filter.conf
-codefilterconfdir = $(filtersdir)/code
+codefilterconfdir = $(filtersconfdir)/code
 
 graphvizfilter = filters/graphviz/graphviz2png.py
 graphvizfilterdir = $(filtersdir)/graphviz
 graphvizfilterconf = filters/graphviz/graphviz-filter.conf
-graphvizfilterconfdir = $(filtersdir)/graphviz
+graphvizfilterconfdir = $(filtersconfdir)/graphviz
 
 musicfilter = filters/music/music2png.py
 musicfilterdir = $(filtersdir)/music
 musicfilterconf = filters/music/music-filter.conf
-musicfilterconfdir = $(filtersdir)/music
+musicfilterconfdir = $(filtersconfdir)/music
 
 sourcefilterconf = filters/source/source-highlight-filter.conf
-sourcefilterconfdir = $(filtersdir)/source
+sourcefilterconfdir = $(filtersconfdir)/source
 
 latexfilter = filters/latex/latex2png.py
 latexfilterdir = $(filtersdir)/latex
@@ -144,8 +146,15 @@
 		chmod +x $$f; \
 	done
 
+fixdatapath:
+	@for f in $(prog); do \
+		echo "Fixing DATA_DIR in $$f"; \
+		$(SED) "s#^DATA_DIR = '.*'#DATA_DIR = '$(ASCIIDOCDATA)'#; s#^DATA_DIR=.*#DATA_DIR=$(ASCIIDOCDATA)#" $$f > $$f.out; \
+		mv $$f.out $$f; \
+	done
+
 install-vim:
-	@for d in $(DESTDIR)/$(vimdir); do \
+	@for d in $(DESTDIR)/$(vimdir) /etc/vim; do \
 		if ! test -d $$d; then continue; fi ; \
 		echo "installing Vim files in $$d" ; \
 		$(INSTALL) -d $$d/syntax ; \
@@ -163,7 +172,7 @@
 	done
 
 
-build: fixconfpath $(manp)
+build: fixconfpath $(manp) fixdatapath
 
 
 install: all $(PROGTARGETS) $(DATATARGETS) progsymlink install-vim
